dim/lib/dim/exporter/raw_rst.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+require 'active_support'
+require 'active_support/core_ext/string/indent'
+require 'erb'
+
+require_relative 'exporterInterface'
+
+module Dim
+  # Exports the requirements to a raw RST format. It doesn't perform
+  # transformations to the requirements' text and it doesn't use any custom RST
+  # directives.
+  # The output is fully compatible with any RST template provided that the text
+  # in the requirements is also valid RST code.
+  class RawRST < ExporterInterface
+    EXPORTER['raw-rst'] = self
+
+    # Path to the directory where the ERB templates are stored.
+    TEMPLATE_DIRECTORY = Pathname.new(__dir__) / 'templates' / 'raw_rst'
+
+    # Path to the ERB template used for the index.
+    INDEX_TEMPLATE = TEMPLATE_DIRECTORY / 'index.rst.erb'
+
+    # Path to the ERB template used for the document.
+    DOCUMENT_TEMPLATE = TEMPLATE_DIRECTORY / 'document.rst.erb'
+
+    # Path to the ERB template used for each requirement.
+    REQUIREMENT_TEMPLATE = TEMPLATE_DIRECTORY / 'requirement.rst.erb'
+
+    # @return [String] The extension that should be used for the files generated
+    #   for this exporter.
+    def self.file_extension
+      'rst'
+    end
+
+    # @param [Dim::Loader] loader The loader used to load the requirements.
+    def initialize(loader)
+      super
+      @hasIndex = true
+    end
+
+    # @param [StringIO] buffer The buffer the content should be written to.
+    # @param [String] category The current category, for example: 'software',
+    #   'architecture', etc.
+    # @param [String] origin The origin of the requirements, for example 'ESR Labs'
+    # @param [Array<String>] modules The name of the modules for the current
+    #   category / origin pair.
+    def index(buffer, category, origin, modules)
+      buffer << ERB.new(INDEX_TEMPLATE.read).result_with_hash(
+        category: category, origin: origin, modules: modules,
+      )
+    end
+
+    # @param [StringIO] buffer The buffer the content should be written to.
+    # @param [String] module_name The name of the current module.
+    def document(buffer, module_name)
+      buffer << ERB.new(document_template_content).result_with_hash(
+        module_name: module_name
+      ) << "\n"
+    end
+
+    # @param [StringIO] buffer The buffer the content should be written to.
+    # @param [Dim::Requirement] requirement The requirement to render.
+    def requirement(buffer, requirement)
+      buffer << ERB.new(requirement_template_content).result_with_hash(
+        requirement: requirement, requirement_id: requirement.id,
+        asil: requirement.asil == "not_set" ? nil : requirement.asil,
+        text: requirement_text(requirement),
+        references: reference_links(requirement.refs),
+        back_references: reference_links(requirement.backwardRefs)
```

Comment:
In newer versions of dox, we use the terms upstream and downstream references.

Reply:
Changed as suggested.